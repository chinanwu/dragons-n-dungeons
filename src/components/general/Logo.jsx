import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';

import './Logo.less';

export const Logo = ({ theme }) => (
  <img
    className="Logo"
    src={require('../../images/dndlogo' + theme + 'V2.png')}
    alt={'logo'}
  />
);

Logo.propTypes = {
  theme: PropTypes.string,
};

export const mapStateToProps = ({ theme }) => ({
  theme: theme.theme,
});

export default connect(mapStateToProps)(Logo);
