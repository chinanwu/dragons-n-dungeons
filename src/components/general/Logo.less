.Logo {
  max-height: 50px;
  padding-left: 15px;
  padding-right: 15px;
  padding-bottom: 10px;
}
