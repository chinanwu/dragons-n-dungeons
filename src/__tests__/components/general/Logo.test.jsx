import { shallow } from 'enzyme';
import React from 'react';

import { Logo, mapStateToProps } from '../../../components/general/Logo.jsx';

jest.unmock('../../../components/general/Logo.jsx');

describe('Logo component', () => {
  describe('rendering', () => {
    it('renders Logo', () => {
      const wrapper = shallow(<Logo />);
      expect(wrapper.getElement()).toMatchSnapshot();
    });
  });

  describe('mapStateToProps', () => {
    it('returned mapped properties', () => {
      const theme = 'light';
      const state = { theme };
      expect(mapStateToProps({ theme: state })).toEqual({ theme: theme });
    });
  });
});
