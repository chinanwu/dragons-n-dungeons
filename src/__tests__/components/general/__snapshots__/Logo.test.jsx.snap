// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component rendering renders Logo 1`] = `
<img
  alt="logo"
  className="Logo"
  src="test-file-stub"
/>
`;
